source/routing/next_stop_time.h

```diff
@@ -216,9 +234,17 @@ struct CachedNextStopTime {
                    const bool clockwise) const;
 
 private:
-    using DtSt = std::pair<DateTime, const type::StopTime*>;
-    IdxMap<JourneyPatternPoint, std::vector<DtSt>> departure;
-    IdxMap<JourneyPatternPoint, std::vector<DtSt>> arrival;
+    struct Fun {
+        typedef CachedNextStopTimeKey const& argument_type;
+        typedef CachedNextStopTime result_type;
+        const type::Data& data;
+        mutable size_t cache_miss;
```

Comment:
I don't understand you use of cache_miss ?

For me it should be the lru that handle this no ?